### PR TITLE
Bump test OpenMPI version to 4.1.4.

### PR DESCRIPTION
**Pull request by romerojosh**
Update test containers to use a newer OpenMPI version (4.1.4).

### CODE REVIEW

File: Dockerfile.test.cpu [hunk 1]
```diff
@@ -98,8 +98,10 @@ RUN pip install --no-cache-dir ray==1.7.0
 
 # Install MPI.
 RUN if [[ ${MPI_KIND} == "OpenMPI" ]]; then \
-        wget --progress=dot:mega -O /tmp/openmpi-3.0.0-bin.tar.gz https://github.com/horovod/horovod/files/1596799/openmpi-3.0.0-bin.tar.gz && \
-            cd /usr/local && tar -zxf /tmp/openmpi-3.0.0-bin.tar.gz && ldconfig && \
+        wget --progress=dot:mega -O /tmp/openmpi-4.1.4-bin.tar.gz https://download.open-mpi.org/release/open-mpi/v4.1/openmpi-4.1.4.tar.gz && \
+            cd /tmp && tar -zxf /tmp/openmpi-4.1.4-bin.tar.gz && \
+            mkdir openmpi-4.1.4/build && cd openmpi-4.1.4/build && ../configure --prefix=/usr/local && \
+            make -j all && make install && ldconfig && \
```
Comment (kvignesh1420) [inline Dockerfile.test.cpu:104]: @romerojosh I was using the same set of commands to build the image with OpenMPI 4.1.4 on my mac. However, when I start a container with that image and run tests I am facing the illegal instruction issue as mentioned in https://github.com/horovod/horovod/pull/3662#issuecomment-1228885412 . The error is surprising as we are not facing it in the CI runs.

Comment (kvignesh1420) [reply Dockerfile.test.cpu:104]:
```ipython
In [1]: import torch
Extension horovod.torch has not been built: /horovod/horovod/torch/mpi_lib_v2.cpython-38-x86_64-linux-gnu.so not found
If this is not expected, reinstall Horovod with HOROVOD_WITH_PYTORCH=1 to debug the build error.
Warning! MPI libs are missing, but python applications are still available.
```

**Comment by EnricoMi** [comment]
@romerojosh there is `openmpi-3.0.0-bin.tar.gz` mentioned also in these files:
- docker/horovod/Dockerfile
- docker/horovod-cpu/Dockerfile

Even `openmpi-4.0.0-bin.tar.gz` is mentioned:
- docs/troubleshooting.rst

Can you update those places as well?

File: Dockerfile.test.cpu (hunk 1 above)
Comment (EnricoMi) [reply Dockerfile.test.cpu:104]: > @romerojosh I was using the same set of commands to build the image with OpenMPI 4.1.4 on my mac.

Our CI tests Horovod on mac with OpenMPI 4.1.1_1-1. All it does is: `brew install -f openmpi`:
https://github.com/horovod/horovod/runs/8077816985?check_suite_focus=true#step:3:111

@kvignesh1420 do install Horovod with `HOROVOD_WITH_MPI=1`? This tells Horovod to check that it compiles against MPI of fail if it cannot find it: https://horovod.readthedocs.io/en/stable/install_include.html#mpi

File: docs/troubleshooting.rst [hunk 2]
```diff
@@ -408,8 +408,8 @@ We recommended reinstalling Open MPI with the ``--enable-orterun-prefix-by-defau
 
 .. code-block:: bash
 
-    $ wget https://www.open-mpi.org/software/ompi/v4.0/downloads/openmpi-4.0.0.tar.gz
-    $ tar zxf openmpi-4.0.0.tar.gz
+    $ wget https://www.open-mpi.org/software/ompi/v4.0/downloads/openmpi-4.1.4.tar.gz
+    $ tar zxf openmpi-4.1.4.tar.gz
     $ cd openmpi-4.0.0
```
Comment (EnricoMi) [inline docs/troubleshooting.rst:413]:
```suggestion
    $ cd openmpi-4.1.4
```

File: Dockerfile.test.cpu (hunk 1 above)
Comment (kvignesh1420) [reply Dockerfile.test.cpu:104]: @EnricoMi I am using the `Dockerfile.test.cpu` to build an image with OpenMPI4.1.4 on my mac. Running the test within that container throws the Illegal instruction error as per : https://github.com/horovod/horovod/pull/3662#issuecomment-1228885412

Didn't face this issue with OpenMPI4.0.0 though!

**Comment by romerojosh** [comment]
@EnricoMi It seems this is just failing due to TF head. Do you know if the upstream issue with the `stream.h` header was resolved?

**Review by EnricoMi — APPROVED** [review]
LGTM!

**Comment by EnricoMi** [comment]
Yes, I am on the TF head issue. They fixed it a few days ago but the fix does not materialize in the nightly file. I'll check with them.

**Comment by EnricoMi** [comment]
Strange, CI thinks this PR removes the following tests:
```
test.integration.test_spark_keras.SparkKerasTests ‑ test_calculate_shuffle_buffer_size
test.integration.test_spark_keras.SparkKerasTests ‑ test_calculate_shuffle_buffer_size_small_row_size
test.integration.test_spark_lightning.SparkLightningTests ‑ test_calculate_shuffle_buffer_size
test.integration.test_spark_lightning.SparkLightningTests ‑ test_calculate_shuffle_buffer_size_small_row_size
test.integration.test_spark_torch.SparkTorchTests ‑ test_calculate_shuffle_buffer_size
test.integration.test_spark_torch.SparkTorchTests ‑ test_calculate_shuffle_buffer_size_small_row_size
```
Can you rebase please to double check?

**Comment by EnricoMi** [comment]
Tests are all green, happy to merge?

**Pull request merged**